config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +110,171 @@ const (
 	// UserDefinedNamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
 	UserDefinedNamespaceIsolated UserDefinedMode = "NamespaceIsolated"
 )
+
+// alertmanagerConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+//
+// +union
+// +kubebuilder:validation:XValidation:rule="self.deploymentMode == 'Deployed' ? has(self.deployed) : (self.deploymentMode == ”) ? !has(self.deployed) : false",message="deployed must be set when deploymentMode is Deployed, and must be unset otherwise"
+type AlertmanagerConfig struct {
+	// deploymentMode determines whether the default Alertmanager instance should be deployed
+	// as part of the monitoring stack.
+	// Allowed values are Deployed and omitted.
+	// When set to Deployed, the Cluster Monitoring Operator
+	// ensures that an Alertmanager instance is created and managed in the `openshift-monitoring` namespace.
+	// Use this field if you want to explicitly opt in to running a platform-level Alertmanager.
+	// When omitted, this means no opinion and the platform is left to choose a default which is subject to change over time.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum="";Deployed
```

Comment:
From our discussion in slack, it sounded like the intention is to have omitted (`""`) be semantically different from `NotDeployed` - are you still wanting to allow users to state that Alertmanager should not be deployed on the cluster by the Cluster Monitoring Operator?

Reply:
no, this come from this bool:
`	// A Boolean flag that enables or disables the main Alertmanager instance
	// in the `openshift-monitoring` namespace.
	// The default value is `true`.
	Enabled *bool `json:"enabled,omitempty"``

Reply:
I'm getting confused.

Should users be allowed to tell the Cluster Monitoring Operator not to deploy Alertmanager?

Reply:
right https://github.com/openshift/cluster-monitoring-operator/blob/34568a9c7f7a0eeb1e8948d03cfc4049adf72362/pkg/manifests/types.go#L91
This is why we have a boolean value and convert it to 'Deployed' and 'Not Deployed'; otherwise, it doesn't make sense